### PR TITLE
chore: refactor git code

### PR DESCRIPTION
**Pull request by IAmAnubhavSaini**
## Description

- Rename variables for better readability.
- Extract code from bigger component.
- Improve boolean logic
- Extract logic out of component
- Unit tests

## Type of change

- Refactoring

## How Has This Been Tested?

Unit tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: git-enhance-1 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.61 **(0.02)** | 38.54 **(0.17)** | 35.89 **(0.03)** | 56.86 **(0.02)**
 :green_circle: | app/client/src/pages/Editor/gitSync/utils.ts | 96.55 **(1.1)** | 77.78 **(11.11)** | 85.71 **(2.38)** | 95.45 **(2.12)**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/GitChangesList.tsx** | **72.73** | **77.54** | **54.17** | **72.22**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/components/GitChangesList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { GitStatusData } from "../../../../reducers/uiReducers/gitSyncReducer";
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/gitSync/components/GitChangesList.test.tsx:1]: Could be `"reducers/uiReducers/gitSyncReducer"`

**Review by ankitakinger — APPROVED** [review]
LGTM

**Comment by IAmAnubhavSaini** [comment]
Cypress pass on second attempt.

Screenshot 2022-05-12 at 1 35 25 PM

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**